### PR TITLE
solveset_doc: add comments and fix LaTeX rendering

### PR DESCRIPTION
**Pull request by hargup**

### CODE REVIEW

**Comment by hargup** [comment]
It will be easier for you to see the changes if you use `git-diff` with `--word-diff` flag.

**Comment by aktech** [comment]
BTW, I think we should further simplify the document, like 
- using `=` instead of `==`,
- using `{-2, 2}` instead of `FiniteSet(-2, 2)`
- using   `{x | x ∊ [0, 2⋅π] ∧ sin(x) = 0}` instead of  </br>
  `ConditionSet(Lambda(x, Eq(sin(x), 0)), [0, 2*pi])` & etc.

(This is unrelated to this PR, though)
We could use python only in examples, so that it's easier to read.

**EDIT**
Oh, I see now, you have expressed that in a comment.

File: doc/src/modules/solvers/solveset.rst [hunk 1]
```diff
@@ -285,15 +316,17 @@ How are symbolic parameters handled in solveset?
...
- [0, oo) and `-n` is the solution only when `n` belongs to the Interval
- (-oo, 0].
+ It simply means `n` is the solution only when it belongs to the `Interval`
+ `[0, oo)` and `-n` is the solution only when `n` belongs to the `Interval`
+ `(-oo, 0]`.
 
  There are various other cases as well which needs to be addressed, like
  say, solving of `2**x + (a - 2)` for `x` where `a` is a symbolic parameter.
- As of now, It returns the solution as an intersection with S.Reals, which
+ As of now, It returns the solution as an intersection with `mathbb{R}`, which
```
Comment (aktech) [inline doc/src/modules/solvers/solveset.rst:325]: `mathbb{R}` ----> `\mathbb{R}`

File: doc/src/modules/solvers/solveset.rst [hunk 2]
```diff
@@ -47,19 +47,21 @@ Why Solveset?
...
-  the set of all solutions that is {2⋅n⋅ⅈ⋅π | n ∊ ℤ}, whereas if x is to be
-  solved in the Real `domain` then only {0} is returned.
+  the set of all solutions that is `\{2 n i \pi \| n \in \mathbb{Z}\}`, whereas if `x` is to be
+  solved in the Real `domain` then only `\{0\}` is returned.
 
 
 Why do we use Sets as an output type?
 -------------------------------------
 
+.. First give an overview of the Sets module of SymPy
```
Comment (aktech) [inline doc/src/modules/solvers/solveset.rst:63]: BTW, Why does comments stops rendering of further text in the section?

**Comment by aktech** [comment]
Thanks for this PR @hargup

File: doc/src/modules/solvers/solveset.rst [hunk 3]
```diff
@@ -233,6 +257,9 @@ How does solveset ensures that it is not returning any wrong solution?
 
  Though, there still a lot of work needs to be done in this regard.
 
+.. Also mention the search based solver we are trying to implement
+.. and the step by step solutions.
```
Comment (aktech) [inline doc/src/modules/solvers/solveset.rst:261]: Should this be written in the future developments sections?

**Comment by aktech** [comment]
Merging this, will fix the typo after merge in https://github.com/sympy/sympy/pull/9500
Thanks @hargup

**Pull request merged**